x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -217,7 +217,7 @@ private XPackSettings() {
         settings.add(PASSWORD_HASHING_ALGORITHM);
         settings.add(INDEX_LIFECYCLE_ENABLED);
         settings.add(SNAPSHOT_LIFECYCLE_ENABLED);
-        settings.add(DATA_FRAME_ENABLED);
```

Comment:
If you agree it's not too hard to provide BWC for this then this line should stay with a TODO to remove it before 8.0.0 release.